docs/data/data-grid/column-menu/ReuseColumnMenuGrid.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import IconFilter from '@mui/icons-material/FilterAlt';
+import SettingsApplicationsIcon from '@mui/icons-material/SettingsApplications';
+import {
+  DataGrid,
+  GridColumnMenuDefault,
+  GridColumnMenuProps,
+  GridColumnMenuItemProps,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+function CustomFilterItem(props: GridColumnMenuItemProps) {
+  const { onClick } = props;
+  return (
+    <Button sx={{ m: 1 }} onClick={onClick} startIcon={<IconFilter />}>
+      Show Filters
+    </Button>
+  );
+}
+
+function CustomUserItem(props: GridColumnMenuItemProps) {
+  const { myCustomHandler, myCustomValue } = props;
+  return (
+    <Button
+      sx={{ m: 1 }}
+      onClick={myCustomHandler}
+      startIcon={<SettingsApplicationsIcon />}
+    >
+      {myCustomValue}
+    </Button>
+  );
+}
+
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      components={{
+        // Override slot for `ColumnMenuFilterItem`
+        ColumnMenuFilterItem: CustomFilterItem,
+        // Hide `ColumnMenuColumnsItem`
+        ColumnMenuColumnsItem: null,
+        // Add new item
+        ColumnMenuUserItem: CustomUserItem,
+      }}
+      componentsProps={{
+        // Swap positions of filter and sort items
+        columnMenuFilterItem: {
+          displayOrder: 0, // Previously `10`
+        },
+        columnMenuSortItem: {
+          displayOrder: 10, // Previously `0`
+        },
+        columnMenuUserItem: {
+          // set `displayOrder` for new item
+          displayOrder: 15,
+          // pass additional props
+          myCustomValue: 'Do custom action',
+          myCustomHandler: () => alert('Custom handler fired'),
+        },
+      }}
+      initialItems={['ColumnMenuUserItem']}
```

Comment:
What is the purpose of `initialItems` prop? I'm not sure I fully understand it

Reply:
As all the default items are injected from respective hooks, user has to specify which of the custom items (that they registered using `components` prop) they want to be displayed. Basically, to control the visibility of custom/user items.

```tsx
<GridColumnMenuDefault
  components={{
    SomeNewItem: SomeComponent,
  }}
  initialItems={[
    // add `SomeNewItem` to the column menu
    'SomeNewItem'
  ]}
/>
```

The purpose of not implicitly adding all the custom items users passes is to allow them to conditionally control the visibility of those items.

Reply:
May be making it `customItems` as @m4theushw suggested makes it more understandable

Reply:
> The purpose of not implicitly adding all the custom items users passes is to allow them to conditionally control the visibility of those items.

Do you have an example with conditional control? Curious to see the use case for it :)

Reply:
Actually I think you are right, we can implicitly add the custom items passed in `components` to the menu and the user can simply remove them or set to `null` if not needed.
Most of the conditional rendering will be happening inside the passed component anyways.